iio_utils.c

```diff
@@ -1,7 +1,21 @@
 #include "iio_utils.h"
+
 #include <string.h>
+#include <ctype.h>
+
+static gint iio_chn_cmp_by_name(gconstpointer ptr_a, gconstpointer ptr_b)
+{
+	struct iio_channel *ch_a = *(struct iio_channel **)ptr_a;
+	struct iio_channel *ch_b = *(struct iio_channel **)ptr_b;
+
+	const char *name_a = iio_channel_get_name(ch_a) ?: iio_channel_get_id(ch_a);
+        const char *name_b = iio_channel_get_name(ch_b) ?: iio_channel_get_id(ch_b);
+
+        return str_natural_cmp(name_a, name_b);
```

Comment:
can `name_a` or `name_b` be NULL?
`str_natural_cmp` does not seem to handle NULL; should it?

Reply:
name_a and name_b can't be NULL - I will document this
str_natural_cmp should not handle NULL

Reply:
> str_natural_cmp should not handle NULL

`str_natural_cmp()` maybe shouldn't
but what about `iio_chn_cmp_by_name()` ?

Reply:
Which pointer did you have in mind? The function parameters (ptr_a, ptr_b) or the data that they point to? (ch_a, ch_b)

Reply:
so, if you would do:
```
if (name_a == name_b)
   return 0;
```
this handles the NULL case for both;

and if you do:
```
if (!name_a)
     return -strlen(name_b);

if (!name_b)
     return strlen(name_a);
```

then there is something that works for the other NULL cases;
i am not sure where this looks best; `iio_chn_cmp_by_name()` or `str_natural_cmp()`; wherever this is put, it does look like a good idea to handle NULL

Reply:
This suggestion could be done and work but what it also does is hide programming errors.
str_natural_cmp() compares strings as the name suggests. Comparing a string with NULL doesn't make sense.
iio_chn_cmp_by_name() is designed to work on a collection (GArray). If by some reason the array doesn't contain valid data, iio_chn_cmp_by_name() should not crash and signal a warning which currently does. But shouldn't handle somebody else's error.

Reply:
Fine by me